templates/commands/describe/describe.go

```diff
@@ -0,0 +1,131 @@
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/abcxyz/abc/templates/common"
+	"github.com/abcxyz/abc/templates/common/specutil"
+	"github.com/abcxyz/abc/templates/common/templatesource"
+	"github.com/abcxyz/abc/templates/model/decode"
+	spec "github.com/abcxyz/abc/templates/model/spec/v1beta1"
+	"github.com/abcxyz/pkg/cli"
+)
+
+type Command struct {
+	cli.BaseCommand
+	flags DescribeFlags
+
+	testFS   common.FS
+	attrList []string
+}
+
+// Desc implements cli.Command.
+func (c *Command) Desc() string {
+	return "show the description and inputs of a given template."
+}
+
+func (c *Command) Help() string {
+	return `
+Usage: {{ COMMAND }} [options] <source>
+
+The {{ COMMAND }} command describe the given template.
+
+The "<source>" is the location of the template to be instantiated. Many forms
+are accepted:
+
+- A remote GitHub or GitLab repo with either a version @tag or with the magic
+    version "@latest". Examples:
+    - github.com/abcxyz/abc/t/rest_server@latest
+    - github.com/abcxyz/abc/t/rest_server@v0.3.1
+- A local directory, like /home/me/mydir
+- (Deprecated) A go-getter-style location, with or without ?ref=foo. Examples:
+    - github.com/abcxyz/abc.git//t/react_template?ref=latest
+	- github.com/abcxyz/abc.git//t/react_template
+`
+}
+
+func (c *Command) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+	c.flags.Register(set)
+	return set
+}
+
+type runParams struct {
+	fs     common.FS
+	stdout io.Writer
+
+	describeTempDirBase string
+}
+
+func (c *Command) Run(ctx context.Context, args []string) error {
+	if err := c.Flags().Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+	fSys := c.testFS
+	if fSys == nil {
+		fSys = &common.RealFS{}
+	}
+	return c.realRun(ctx, &runParams{
+		fs:     fSys,
+		stdout: c.Stdout(),
+	})
+}
+
+// readRun provides fakable interface to test Run.
+func (c *Command) realRun(ctx context.Context, rp *runParams) (rErr error) {
+	var tempDirs []string
+	defer func() {
+		for _, d := range tempDirs {
+			rErr = errors.Join(rErr, rp.fs.RemoveAll(d))
+		}
+	}()
+
+	_, templateDir, err := templatesource.Download(ctx, rp.fs, rp.describeTempDirBase, c.flags.Source, c.flags.GitProtocol)
+	if templateDir != "" { // templateDir might be set even if there's an error
+		tempDirs = append(tempDirs, templateDir)
+	}
+	if err != nil {
+		return err //nolint:wrapcheck
+	}
+
+	specPath := filepath.Join(templateDir, specutil.SpacFileName)
+	f, err := rp.fs.Open(specPath)
+	if err != nil {
+		return fmt.Errorf("error opening template spec: ReadFile(): %w", err)
+	}
+	defer f.Close()
+
+	specI, err := decode.DecodeValidateUpgrade(ctx, f, specutil.SpacFileName, decode.KindTemplate)
+	if err != nil {
+		return fmt.Errorf("error reading template spec file: %w", err)
+	}
+
+	spec, ok := specI.(*spec.Spec)
+	if !ok {
+		return fmt.Errorf("internal error: spec file did not decode to spec.Spec")
+	}
+
+	c.attrList = specutil.ParseSpecToList(spec)
```

Comment:
Does this really need to be c.attrList? Could it be a local variable instead (just `attrList`)? Generally we should try to avoid storing state where it's not needed, which increases complexity.

Reply:
the purpose of this is for doing diff in the tests. It doesn't need to be c.attrList, but this makes tests easier.